pkg/sloop/queries/respayloadquery.go

```diff
@@ -91,11 +105,11 @@ func GetResPayload(params url.Values, t typed.Tables, startTime time.Time, endTi
 	return bytes, nil
 }
 
-//todo: add unit tests
 func getSeekKey(keyComparator *typed.WatchTableKey, startTime time.Time) *typed.WatchTableKey {
-	seekKey := keyComparator
-	seekKey.PartitionId = untyped.GetPartitionId(startTime)
-	seekKey.Timestamp = startTime
+	kind := keyComparator.Kind
+	namespace := keyComparator.Namespace
+	name := keyComparator.Name
+	seekKey := typed.NewWatchTableKey(untyped.GetPartitionId(startTime), kind, namespace, name, startTime)
```

Comment:
NewWatchTableKey returns a pointer, right?

Reply:
yes, a pointer to a new instance